tools/update-dll-msvc.bat

```diff
@@ -0,0 +1,57 @@
+@echo off
+rem Update the DLL using MSVC.
+rem If the old DLL is in use, rename it to avoid compilation error.
+rem
+rem usage: update-dll-msvc [arch] [makeopts]
+rem
+rem   [arch] is 32 or 64. If omitted, it is automatically detected from the
+rem   %PROCESSOR_ARCHITECTURE% environment.
+rem   [makeopts] is option(s) for nmake.
+rem
+rem
+rem Sample .vimrc:
+rem
+rem NeoBundle 'Shougo/vimproc.vim', {
+rem \ 'build' : {
+rem \     'windows' : 'tools\\update-dll-msvc',
+rem \     'cygwin' : 'make -f make_cygwin.mak',
+rem \     'mac' : 'make -f make_mac.mak',
+rem \     'linux' : 'make',
+rem \     'unix' : 'gmake',
+rem \    },
+rem \ }
+
+if "%PROCESSOR_ARCHITECTURE%"=="AMD64" (
+    set vimproc_arch=64
+    set cpu_arch=AMD64
+    set msvc_arch=amd64
+) else (
+    set vimproc_arch=32
+    set cpu_arch=x86
+    set msvc_arch=x86
+)
+
+set vimproc_dllname=vimproc_win%vimproc_arch%.dll
+
+
+call "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Auxiliary\Build\vcvarsall.bat" %msvc_arch%
```

Comment:
I'm wondering if it is okay to support only VC 2019. (Shouldn't be the name of the batch file `update-dll-msvc2019.bat`?)
Is the directory name `BuildTools` correct? I think that part depends on the edition of VC. (e.g. `Community`, `Professional`, `Enterprise`)

Reply:
@k-takata , your understanding is right. I will try to update the file update-dll-msvc.bat so that it can automatically to support from VC2008 to VC2019. It will be on a new pull request which I'm working on it.

Reply:
@k-takata , I have just update the file update-dll-msvc.bat and add one excutable file which name is vswhere.exe to support the find the installation path of Visual Studio. Currently I onlyl test it on Visual C++ Build Tools 2019 but not other version of compiler.